lib/fizzy/execute.cpp

```diff
@@ -1229,9 +1230,10 @@ ExecutionResult execute(
             uint32_t ret = static_cast<uint32_t>(cur_pages);
             try
             {
+                // TODO use memory_pages_limit passed to instantiate
                 const size_t memory_max_pages =
```

Comment:
Couldn't we update `instance.memory_pages_limit` with the imported max?

Reply:
I'll try it